flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/UserDefinedFunction.scala

```diff
@@ -24,4 +24,19 @@ package org.apache.flink.table.functions
   * User-defined functions must have a default constructor and must be instantiable during runtime.
   */
 trait UserDefinedFunction {
```

Comment:
The Scala wrapper class can be abstract class. 

MathTrait.scala
```Scala
trait MathTrait {
    def sum(x: Int, y: Int) = x + y
}
```

MathTraitWrapper.scala
```Scala
abstract class MathTraitWrapper extends MathTrait {
}
```

JavaMath.java
```Java
public class JavaMath extends MathTraitWrapper {
	public static void main(String[] args) {
		new JavaMath();
	}

	public JavaMath() {
		System.out.println(sum(2, 2));
	}
}
```

This code works as expected, printing the number 4 when it is run.